### DISCUSSION TITLE
Periodical reconnect

### DISCUSSION THREAD

**Discussion by leshniak** [bubuntux/nordlynx#29] [Ideas]
Shouldn't it has an option for periodical reconnect to another server, to help balance the load?

PS. great job.

**Comment by leshniak** [comment]
Ok, I guess I can do it by `s6-sleep 86400 s6-svc -r /run/s6/services/wireguard &` in `POST_UP`.

**Reply by leshniak** [reply]
> Ok, I guess I can do it by `s6-sleep 86400 s6-svc -r /run/s6/services/wireguard &` in `POST_UP`.

Not exactly, because it doesn't kill the background processes. I've ended up with `POST_UP=s6-sleep 28800 bash -c "kill -SIGTERM -\$$(s6-svstat -p /run/s6/services/wireguard)" &` but it's to hacky without a proper `trap` in the shell script.

**Comment by leshniak** [comment]
maybe
```
sleep infinity & PID=$!
trap "kill $PID; exit" INT TERM
```
instead of https://github.com/bubuntux/nordlynx/blob/main/rootfs/etc/services.d/wireguard/run#L35-L36 ?

**Comment by bubuntux** [comment]
i did a big rebase and added a trap,  so adding  s6-sleep 86400 s6-svc -r /run/s6/services/wireguard & in the POST_UP works fine for me now

**Reply by bubuntux** [reply]
actually it seems to work even without the trap

**Reply by leshniak** [reply]
Thank you! But I think that `sleep infinity` will be still dangling after terminating the script :/. We should terminate it too, somehow. Either by saving its PID and killing it after receiving a SIGTERM (best from the `s6-svc` perspective) or by doing something like `sleep ${RECONNECT_TIME:-infinity}`. Or maybe it should not be sent to the background?

**Reply by leshniak** [reply]
Yes, it works basically, but leaves `sleep infinity` untouched.

**Reply by bubuntux** [reply]
hmmm i see...

**Comment by bubuntux** [comment]
i ended up adding a new variable "REFRESH" that specify the number of seconds to refresh the connection

**Reply by bubuntux** [reply]
actually gonna rename it to reconnect

**Reply by leshniak** [reply]
Sounds good! I forked your project for my own use anyway, because I don't like the Ubuntu bloat ;)

**Reply by bubuntux** [reply]
i was using alpine for that reason but ubuntu makes more sense so i can reuse the 30-module script to increase compatibility and ease of use, but hey that is the beauty of open source, do with it as you wish.

**Reply by leshniak** [reply]
Yup, I just don't need it, I have wireguard already in my kernel and I control it on my own. For me it's better to have a base image 10 times lighter. But I also understand your point, that's why I decided to fork it ;)
